### DISCUSSION TITLE
Can we add original value of creator coin investments in our wallet?

### DISCUSSION THREAD

**Discussion by derishaviar** [deso-protocol/dips#52] [Medium CIPs]
For many creators who are not day traders keeping track of our investment gains and loss is very difficult. The more creator coins we own the more difficult it is to remember how much our original investment was and that ultimately makes it harder to determine which creator coins to sell or hold. 

I think if we add a trade history in our wallet or at least show the amount of original investment (not including diamonds and gifted coin) it would make more people including myself more confident to trade more creator coins. 

I know now it is possible to check on bitcloutsignal manually but having this data right in our wallets would be very helpful and easier to manage/organise. Please consider my idea. Thanks so much :)

**Comment by tijno** [comment]
yes would love this too, but not sure how easy this is to implement. Maybe the new databases being implemented will make it easier.

the amount should probably be an average of all buys.

how should sales be treated though?

eg 
buy 10 @ 100
buy 5 @ 150
=> avr paid 83

sell 7 @ 50

what is the average after the sell?

**Reply by derishaviar** [reply]
Hi @tijno thanks for replying, yes i understand this is not easy and I'm definitely not able to change the code for this myself!

Yes, maybe we could have a Net asset value (NAV) showing beside current market value similar to in mutual funds or if that's too complicated maybe we can hover or click the creator coin in our wallet and it takes us to another page or pop-up with our trade history for that particular creator coin. Something similar to what Binance or any other exchange shows us for the 'order trade history'.

Would it be too complicated?

**Reply by AikonMishima** [reply]
Trade history would indeed be easier, as calculating the gains and loss is not simple due to the bonding curve.

For instance, if you buy
- 1 $derishaviar Coin
- at $100 Coin Price ( with 1 hypothetical $CLOUT at exchange value $100)
- with 0% FR

You will have paid $105.37 , not $100, as the bonding curve is in effect.

Possible solution:
Showing trade history and the current value in $CLOUT will give a better insight in wether your trades have gone up or down in Creator Coin value.

That way, you can see if you made profit or not.
The USD price of $CLOUT can be confusing to those who aren't used to trading.

**Reply by tijno** [reply]
> Showing trade history and the current value in $CLOUT will give a better insight in wether your trades have gone up or down in Creator Coin value.

Yes +1 for this

**Reply by derishaviar** [reply]
> Trade history would indeed be easier, as calculating the gains and loss is not simple due to the bonding curve.
> 
> For instance, if you buy
> 
> * 1 $derishaviar Coin
> * at $100 Coin Price ( with 1 hypothetical $CLOUT at exchange value $100)
> * with 0% FR
> 
> You will have paid $105.37 , not $100, as the bonding curve is in effect.
> 
> Possible solution:
> Showing trade history and the current value in $CLOUT will give a better insight in wether your trades have gone up or down in Creator Coin value.
> 
> That way, you can see if you made profit or not.
> The USD price of $CLOUT can be confusing to those who aren't used to trading.

Oh yes that's true, the bonding curve does make it much harder to calculate. Great solution @AikonMishima Thanks for explaining everything in details!

**Comment by lucefx** [comment]
I don't think you need to average them together? Like with regular trading. Selling is just reducing position size, but doesn't change your average buy-in price.

Also, I hope this makes sense, but why not use an average of 'coins minted' rather than clout or USD? That way regardless of the price of clout, you can tell if your holding is up or down. Eg:

Buy 1 @luce coin when Luce has 3 coins in circulation
Buy 4 @luce coins when Luce has 10 coins in circulation
Total buys = 5 @luce coins
Average coins in circulation of buys = [(1x3)+(4x10)] / 5 = 43/5 = 8.6

If Luce currently has 13 coins in circulation, then your investment is up, if she has 7 coins in circulation then your investment is down.

I don't know the formula for the bonding curve but I know it's not linear, still I'm sure that someone can come up with a formula so that the change of coins minted can be shown as normal % gains/losses for an investor based on their average coins-minted buy-in figure.

**Reply by tijno** [reply]
Thx @lucefx - glad you joined GitHub to contribute :)

Re [Bonding curve](https://docs.bitclout.com/#the-creator-coin-supply-curve)

This is the creator coin formula

`price_in_bitclout=0.003×creator_coins_in_circulation^2`

> If Luce currently has 13 coins in circulation, then your investment is up, if she has 7 coins in circulation then your investment is down.

Does this translate to an easy % up / down metric? 

> use an average of 'coins minted' 

I dont think that makes it simpler.

As per @AikonMishima - this still applies even when using the coins minted calc "calculating the gains and loss is not simple due to the bonding curve"

**Reply by AikonMishima** [reply]
The main sources of fluctuation on the price are
- The creator coin itself (buys / sells fluctuating the market cap & coins in circulation)
- The USD price of $CLOUT

If you remove the influence of the USD exchange rate, and only look at these 2 factors ...
- How much $CLOUT have you paid in the past
- How much $CLOUT is it worth right now

... you will see how much you have gained, or lost.

**Important note:**
Incorporating automatic calculations for the whole wallet will be taxing for the page load (api calls to search for the purchases > remove the previous sales > calculate the difference in $CLOUT > check to see if it's lower or more than the current worth based on the bonding curve).

Therefor, a secondary page with a trade history per Creator Coin will be more beneficial for the user experience.

**Mockup:**
(did this on my phone, quality isn't great)

![History](https://user-images.githubusercontent.com/84419699/125482260-a536c8e7-5d1d-47ec-b926-c36c4741ac93.png)

**Reply by derishaviar** [reply]
> I don't think you need to average them together? Like with regular trading. Selling is just reducing position size, but doesn't change your average buy-in price.
> 
> Also, I hope this makes sense, but why not use an average of 'coins minted' rather than clout or USD? That way regardless of the price of clout, you can tell if your holding is up or down. Eg:
> 
> Buy 1 @Luce coin when Luce has 3 coins in circulation
> Buy 4 @Luce coins when Luce has 10 coins in circulation
> Total buys = 5 @Luce coins
> Average coins in circulation of buys = [(1x3)+(4x10)] / 5 = 43/5 = 8.6
> 
> If Luce currently has 13 coins in circulation, then your investment is up, if she has 7 coins in circulation then your investment is down.
> 
> I don't know the formula for the bonding curve but I know it's not linear, still I'm sure that someone can come up with a formula so that the change of coins minted can be shown as normal % gains/losses for an investor based on their average coins-minted buy-in figure.

Oh that's true... This is more complicated than I thought lol but thanks for the feedback!

So if we use 'coins minted' how would this affect the value if someone has decided to sell the creator coin? Wouldn't that mean the number would go down?

**Reply by derishaviar** [reply]
> The main sources of fluctuation on the price are
> 
> * The creator coin itself (buys / sells fluctuating the market cap & coins in circulation)
> * The USD price of $CLOUT
> 
> If you remove the influence of the USD exchange rate, and only look at these 2 factors ...
> 
> * How much $CLOUT have you paid in the past
> * How much $CLOUT is it worth right now
> 
> ... you will see how much you have gained, or lost.
> 
> **Important note:**
> Incorporating automatic calculations for the whole wallet will be taxing for the page load (api calls to search for the purchases > remove the previous sales > calculate the difference in $CLOUT > check to see if it's lower or more than the current worth based on the bonding curve).
> 
> Therefor, a secondary page with a trade history per Creator Coin will be more beneficial for the user experience.
> 
> **Mockup:**
> (did this on my phone, quality isn't great)
> 
> ![History](https://user-images.githubusercontent.com/84419699/125482260-a536c8e7-5d1d-47ec-b926-c36c4741ac93.png)

@AikonMishima thanks for the very detailed explanation... i am absolutely loving this idea and the mockup!! What do you think would be best shown in the trade history for a simple yet effective UX?

**Reply by AikonMishima** [reply]
I’ll make a mock-up asap.

**Reply by derishaviar** [reply]
> I’ll make a mock-up asap.

great, thanks @AikonMishima you're a legend!

**Reply by lucefx** [reply]
> The main sources of fluctuation on the price are
> 
> * The creator coin itself (buys / sells fluctuating the market cap & coins in circulation)
> * The USD price of $CLOUT
> 
> If you remove the influence of the USD exchange rate, and only look at these 2 factors ...
> 
> * How much $CLOUT have you paid in the past
> * How much $CLOUT is it worth right now
> 
> ... you will see how much you have gained, or lost.
> 
> **Important note:**
> Incorporating automatic calculations for the whole wallet will be taxing for the page load (api calls to search for the purchases > remove the previous sales > calculate the difference in $CLOUT > check to see if it's lower or more than the current worth based on the bonding curve).
> 
> Therefor, a secondary page with a trade history per Creator Coin will be more beneficial for the user experience.
> 
> **Mockup:**
> (did this on my phone, quality isn't great)
> 
> ![History](https://user-images.githubusercontent.com/84419699/125482260-a536c8e7-5d1d-47ec-b926-c36c4741ac93.png)

Honestly, I'd be happy enough with 'average buy price' and 'current creator coin price', both shown in clout. I'll do the maths myself. But we need to be able to look at our wallets, which sometimes have hundreds of coins, and know what's up and what's down, without having to navigate away from the wallet.
